File: submissions/p1-rats/8.rb
def f(i, n, w) 
e = n
m = []
def o(t)
y = " "
for x in 1..t-1
y += " "
end
return y;
end
def drawOneline(h, a, b, c)
line = ""
for x in 0..c-1
line += (a[h]);
h += b
if x < c-1
line += "|"
end
end
return line
end
for x in 0..(i.length/w)
m.push(i[x*w..x*w+w-1])
end
m = m.reject { |c| c.nil? }
if m[-1] == ""
m.delete(m[-1])
end
y = m[-1].length
for x in y..w-1
m[-1] = m[-1] + " "
end
q = m.length / e
l = (m.length) % e
if(l != 0)
q += 1
end
for x in m.length..e*q
m.push(o(w))
end
r = ""
for x in 0..q-1
r += drawOneline(x, m, q, e)
if x != q-1
r+="\n"
end
end
print r
end

File: submissions/p1-rats/8.py
def f(input, n, width) :
	str = input
	str = str.replace(" ", ".")
	
	column = n
	tagging = width
	
	my_list = []
	
	def printme():
		y = "."
		for x in range(1, tagging):
			y += "."
		return y;
		
	def drawOneline(lineNum, array, rowparam):
		line = ""
		for x in range(0, column):
			line += (array[lineNum]);
			lineNum += rowparam
			if x < column-1:
				line += "|"
		return line
	
	for x in range(0, column*column):
	    my_list.append(str[x*tagging:x*tagging+tagging])
	my_list = [x for x in my_list if x != '']
	
	y = len(my_list[-1])
	for x in range(y, tagging):
		my_list[-1] = my_list[-1] + "."
	
	row = len(my_list) / column
	row = int(row)
	leftover = len(my_list) % column
	
	if(leftover != 0):
		row += 1
		
	for x in range(len(my_list), column*row):
			my_list.append(printme())
	
	for x in range(0, row):
		print(drawOneline(x, my_list, row))

	return